src/reactions.js

```diff
@@ -95,27 +30,25 @@ class mangoReactions {
       return totalCount > 0 ? totalCount : 0;
     }
 
-    let result = {};
-    const postID = this.options.postID;
+    const { postID } = this.options;
+    // clone reactions
     const reactions = this.options.reactions.slice();
-    reactions.map((reaction) => {
-      result[reaction] = getTotalCount(postID, reaction);
-      // map to loop but not change anything ;p fix later
-      return reaction;
-    });
+    // generate a object based on reactions array
+    // with the proper count
+    const result = Object.assign(...reactions.map(reaction => ({
+      [reaction]: getTotalCount(postID, reaction),
+    })));
     return result;
   }
   doRequest() {
-    const postID = this.options.postID;
-    const accessToken = this.options.access_token;
-    const query = this.mountRequest();
-    const callback = this.options.callback;
-    const url = `https://graph.facebook.com/v2.8/?ids=${postID}&fields=${query}&access_token=${accessToken}`;
+    const { postID, access_token, callback } = this.options;
+    const query = this.memoizedMountRequest();
+    const url = `https://graph.facebook.com/v2.8/?ids=${postID}&fields=${query}&access_token=${access_token}`;
     fetch(url)
     .then((blob) => blob.json())
     .then((data) => callback(this.parseRequest(data)))
     .catch((error) => {
-      throw error.message;
+      throw new Error(error.message);
```

Comment:
interesting but maybe dangerous, the user would have to be responsible to write a complete callback function to allow error handle and data handle at same time, could be messy.
as far from now it only returns a object, it's easier if the user has little to no experience with those type of things like @raulmangolin AHAHAHAHAHA